### PR TITLE
Flythrough Line Drawing

### PR DESCRIPTION
**Pull request by Jossos**
- Added Line drawing of Selected flythroughs
- Fixes [AB#2144](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/2144)

### CODE REVIEW

File: src/scene/vcFlythrough.cpp [hunk 1]
```diff
@@ -139,6 +140,9 @@ void vcFlythrough::AddToScene(vcState *pProgramState, vcRenderData * /*pRenderDa
       UpdateCameraPosition(pProgramState);
     }
   }
+
+  if (m_selected && m_pLine != nullptr && m_flightPoints.length > 1)
+    pRenderData->lines.PushBack(m_pLine);
```
Comment (OsiJr) [inline src/scene/vcFlythrough.cpp:145]: The line shouldn't appear when recording or playback

**Review by OsiJr — CHANGES_REQUESTED** [review]

File: src/scene/vcFlythrough.cpp [hunk 2]
```diff
@@ -389,3 +396,16 @@ void vcFlythrough::UpdateCameraPosition(vcState *pProgramState)
     }
   }
 }
+
+void vcFlythrough::UpdateLinePoints()
+{
+  if (m_flightPoints.length > 1)
+  {
+    if (m_pLine == nullptr)
+      vcLineRenderer_CreateLine(&m_pLine);
```
Comment (peterAdams1014) [inline src/scene/vcFlythrough.cpp:405]: Missing `vcLineRenderer_DestroyLine()`.

File: src/scene/vcFlythrough.cpp [hunk 3]
```diff
@@ -389,3 +396,16 @@ void vcFlythrough::UpdateCameraPosition(vcState *pProgramState)
...
+    udDouble3 *pPositions = udAllocType(udDouble3, m_flightPoints.length, udAF_Zero);
```
Comment (peterAdams1014) [inline src/scene/vcFlythrough.cpp:406]: `pPositions` leaks.

**Review by peterAdams1014 — CHANGES_REQUESTED** [review]

File: src/scene/vcFlythrough.cpp (hunk 3 above)
Comment (Jossos) [reply src/scene/vcFlythrough.cpp:406]: Fixed

File: src/scene/vcFlythrough.cpp [hunk 4]
```diff
@@ -389,3 +396,17 @@ void vcFlythrough::UpdateCameraPosition(vcState *pProgramState)
     }
   }
 }
+
+void vcFlythrough::UpdateLinePoints()
+{
+  if (m_flightPoints.length > 1)
+  {
+    vcLineRenderer_DestroyLine(&m_pLine);
```
Comment (peterAdams1014) [inline src/scene/vcFlythrough.cpp:404]: What you had before was fine. Just check if its null.

**Review by peterAdams1014 — CHANGES_REQUESTED** [review]

File: src/scene/vcFlythrough.cpp (hunk 4 above)
Comment (Jossos) [reply src/scene/vcFlythrough.cpp:404]: ok. it looks like createLine doesn't free it that's why I did it this way

**Review by peterAdams1014 — DISMISSED** [review]

**Review by samsinsane — CHANGES_REQUESTED** [review]
This has merge conflicts.

**Review by samsinsane — APPROVED** [review]

**Pull request merged**